client/js/store/modules/auth.js

```diff
@@ -24,64 +52,22 @@ export default {
     },
   },
   actions: {
-    requestLogin(context, data) {
-      return new Promise((resolve, reject) => {
-        io.socket.post(
-          '/user/login',
-          {
-            username: data.username,
-            password: data.password,
-          },
-          function handleResponse(resData, jwres) {
-            if (jwres.statusCode === 200) {
-              context.commit('authSuccess', data.username);
-              return resolve();
-            }
-            context.commit('clearAuth');
-            return reject(jwres.body.message);
-          }
-        );
-      });
+    async requestLogin(context, { username, password }) {
+      await handleLogin(context, username, password);
```

Comment:
Maybe this should return so that the dispatched action can await the resulting promise. That way callers can manage loading states. I don't think this function needs to be async or to await at that point
```suggestion
    requestLogin(context, { username, password }) {
      return handleLogin(context, username, password);
```